### PR TITLE
I15 oc mapping

### PR DESCRIPTION
**Pull request by icolwell**
**Summary**
Everything was moved from "matlab_simulation/code v1.0/ME597-6-Mapping/" to "matlab_simulation/06-mapping/occupancy_grid/" or to the appropriate folder. Some duplicated functions were also removed.

**Examples**
There are 4 examples I included:

%% Occupancy grid mapping example 1
Uses a 50 x 60 cell map with a laser scanner, map is updated using Bresenham ray trace mode.

%% Occupancy grid mapping example 2 Uses a 50 x 60 cell map with a laser scanner, map is updated using windowed block update mode.

%% Occupancy grid mapping example 3
Uses a 50 x 60 cell map with sonar, map is updated using windowed block update mode.

%% Occupancy grid mapping example 4
Performance test on a large map. Uses a 1000 x 1000 cell map with a laser scanner, map is updated using Bresenham ray trace mode. Laser scanner range and FOV increased.

**New Functions:**
-ogmap = Does occupancy grid mapping
-get_sonar_range = returns the range of nearest obstacle using sonar
-load_cell_map = generates a couple different types of cell maps
-inverse_scanner-window = does block update mode but only iterates over the window defined by the max range of the sensor and boundaries of the map. Could likely be optimized further based on FOV and sensor angle but it is a good start.
-various plotting functions specific to these mapping examples

**UPDATES AFTER CODE REVIEW**
-Added the map border padding as Stan mentioned to remove an if statement from a loop.
-Reviewed all files and made small formatting improvements according to Stan's comment.
-Updated inverse_scanner_bres.m to take in the probabilities of occupied and unoccupied cells as arguments.
-Clarified a couple comments

### CODE REVIEW

File: matlab_simulation/04-sensor/get_sonar_range.m [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    for j = 1:w_N
+        % Map coordinates of window loop cell
+        x_wi = x(1) - win_pos(1) + i;
+        y_wj = x(2) - win_pos(2) + j;
+        
+        % Check if cell is occupied 
+        if (map(x_wi, y_wj) == 0)
+            % Check if edge map (edge of map is considered occupied)
+            if ~(x_wi == 1 || x_wi == M || y_wj == 1 || y_wj == N)
+                continue
```
Comment (StanJBrown) [inline matlab_simulation/04-sensor/get_sonar_range.m]: You can avoid having an extra check in your code if you pad all the outer cells with zeros. Then this loop becomes one single if statement. http://www.mathworks.com/help/images/ref/padarray.html

I did this in my pull request. #35 if you want an example. Generally minimizing the number of if statements in a loop greatly increases the code run time in matlab.

File: matlab_simulation/04-sensor/inverse_scanner_bres.m [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+% Bound the robot within the map dimensions
+x1 = max(1,min(M,round(x)));
+y1 = max(1,min(N,round(y)));
+
+% Calculate position of measured object (endpoint of ray)
+endpt = [x y] + r*[cos(theta) sin(theta)];
+
+% Bound the endpoint within the map dimensions
+x2 = max(1,min(M,round(endpt(1))));
+y2 = max(1,min(N,round(endpt(2))));
```
Comment (StanJBrown) [inline matlab_simulation/04-sensor/inverse_scanner_bres.m]: Small syntax comment, try to put spaces after each comma. It generally helps with the readability.

File: matlab_simulation/06-mapping/occupancy_grid/plot_occupancy_grid.m [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+% Output:
+% 	Plot window showing occupancy grid 
+
+figure(fig);
+clf; 
+hold on;
+image(100*(og));
+colormap(gray);
+axis([0 N 0 M])
+title('Current occupancy grid map')
```
Comment (StanJBrown) [inline matlab_simulation/06-mapping/occupancy_grid/plot_occupancy_grid.m:17]: Nice job making the plotting into functions

**Comment by icolwell** [comment]
:smile:

**Pull request merged**